src/tf_deadness_analysis.cc

```diff
@@ -427,6 +452,15 @@ bool DeadnessAnalysisImpl::HasInputsWithMismatchingDeadness(const Node& node) {
   Predicate* pred = nullptr;
   for (const Edge* edge : node.in_edges()) {
     auto it = predicate_map_.find(InputEdgeToTensorId(edge));
+    if (it != predicate_map_.end()) {
+      NGRAPH_VLOG(5) << "Cannot find predicate for Edge ";
+      NGRAPH_VLOG(5) << "Src " << edge->src()->name() << "["
+                     << edge->src()->type_string() << "]"
+                     << " ,Src Idx " << edge->src_output() << " DST "
+                     << edge->dst()->name() << "[" << edge->dst()->type_string()
+                     << "]"
+                     << " ,Dst Idx " << edge->dst_input();
+    }
     CHECK(it != predicate_map_.end());
```

Comment:
There are 3 instances of `predicate_map_.find` in the tf_deadness code. The 3rd one is here:
https://github.com/NervanaSystems/ngraph-tf/blob/master/src/tf_deadness_analysis.cc#L460

These are great/useful prints, and help a lot in debugging. But you have added them in 2 of the 3 places for instance. In the spirit of DRY, and also to cover the 3rd instance, what if we have a function in DeadnessAnalysisImpl

```
appropriate_type findInMapOrFail(const TensorId& tensor_id){
    auto it = predicate_map_.find(tensor_id);
    if (it != predicate_map_.end()) {the_debug_prints_you_added;}
    CHECK(it != predicate_map_.end());   //This i suppose exits the program?
    return it->second;
}
```

Reply:
I did not add it here , as the check prints the TensorID. There is no way of getting Edge from the TensorId

Reply:
The changes here were just to have some logs , as we were seeing many crashes. We have a Ticket to Update this code. May be i can PR there, if it is required? It is repeated once .

Reply:
Ah, makes sense. In which case, plz ignore my above comment